lib/InternalJavaScript/04-AsyncIterator.js

```diff
@@ -0,0 +1,45 @@
+function initAsyncIterators() {
+    function _makeAsyncIterator(iterable, asyncIterMethod, syncIterMethod) {
+        if (asyncIterMethod) {
+            return asyncIterMethod.call(iterable);
+        }
+
+        let syncIterator = syncIterMethod.call(iterable);
```

Comment:
Proper block scoping is still behind a flag in SH, so let's avoid the behavior changing here unless necessary and just use `var` for now (everywhere in this file)